File: frontend/src/types/vehicle.ts
export type VehicleStatus = 'ACTIVE' | 'IN_SERVICE' | 'INACTIVE' | 'SOLD' | 'DECOMMISSIONED';
export type VehicleType = 'CAR' | 'VAN' | 'TRUCK' | 'MOTORCYCLE' | 'OTHER';
export type FuelType = 'DIESEL' | 'PETROL' | 'HYBRID' | 'ELECTRIC' | 'LPG' | 'OTHER';
export type OwnershipType = 'OWNED' | 'LEASED' | 'RENTED' | 'OTHER';

export interface VehicleRequest {
  /** Target organization. Required for SUPERADMIN-created vehicles, omitted otherwise. */
  businessId?: number;
  licensePlate: string;
  vin: string;
  brand: string;
  model: string;
  manufactureYear: number;
  vehicleType: VehicleType;
  fuelType: FuelType;
  ownershipType: OwnershipType;
  status: VehicleStatus;
  department: string;
  currentMileage: number;
}

export interface Vehicle extends Omit<VehicleRequest, 'businessId'> {
  id: number;
  businessId: number | null;
  assignedUserId: number | null;
  assignedDriverName: string | null;
  imageUrl: string | null;
  imageOriginalFileName: string | null;
  createdAt: string;
  updatedAt: string;
}

export interface VehicleFilters {
  status?: string;
  vehicleType?: string;
  fuelType?: string;
  ownershipType?: string;
  department?: string;
  assignedUserId?: number;
  licensePlate?: string;
  businessId?: number;
}

export interface VehicleAssignmentRequest {
  assignedUserId: number | null;
  /** Sent for compatibility with older fleet-service deployments. */
  assignedDriverName?: string | null;
  /** Sent so older fleet-service deployments preserve the department. */
  department?: string | null;
}

File: fleet-service/src/main/java/com/fleet/fleet/entity/Vehicle.java
package com.fleet.fleet.entity;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.PrePersist;
import jakarta.persistence.PreUpdate;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.time.Instant;

@Entity
@Table(name = "vehicles")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Vehicle {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column
    private Long businessId;

    @Column(nullable = false, length = 32)
    private String licensePlate;

    @Column(length = 64)
    private String vin;

    @Column(nullable = false, length = 80)
    private String brand;

    @Column(nullable = false, length = 80)
    private String model;

    private Integer manufactureYear;

    @Enumerated(EnumType.STRING)
    @Column(length = 30)
    private VehicleType vehicleType;

    @Enumerated(EnumType.STRING)
    @Column(length = 30)
    private FuelType fuelType;

    @Enumerated(EnumType.STRING)
    @Column(length = 30)
    private OwnershipType ownershipType;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false, length = 30)
    @Builder.Default
    private VehicleStatus status = VehicleStatus.ACTIVE;

    @Column(length = 120)
    private String department;

    private Long assignedUserId;

    @Column(length = 160)
    private String assignedDriverName;

    private Long currentMileage;

    @Column(nullable = false, updatable = false)
    private Instant createdAt;

    @Column(nullable = false)
    private Instant updatedAt;

    @PrePersist
    void onCreate() {
        Instant now = Instant.now();
        createdAt = now;
        updatedAt = now;
        if (status == null) {
            status = VehicleStatus.ACTIVE;
        }
    }

    @PreUpdate
    void onUpdate() {
        updatedAt = Instant.now();
    }
}
